draft-ietf-suit-trust-domains.md

```diff
@@ -372,6 +376,14 @@ A dependency prefix can be used with a component identifier. This allows complex
 
 A dependency prefix can also be used to indicate when a dependency manifest needs to be processed by a secondary manifest processor, as described in {{hierarchical-interpreters}}.
 
+# Uninstall {#suit-uninstall}
+
+In some systems, particularly with multiple, independent, optional components, it may be that there is a need to uninstall the components that have been installed by a manifest. Where this is expected, the uninstall command sequence can provide the sequence needed to cleanly remove the components defined by the manifest and its dependencies. In general, suit uninstall will contain primarily unlink directives.
+
+WARNING: This can cause faults where there are loose dependencies (e.g. version-only, see {{I-D.ietf-suit-update-management}}), since a component can be removed while it is depended upon by another component.
```

Comment:
Since it's an "unlink" not a "removal", what is meant by a "loose" dependency?  Is that one that does not hold a link reference?   Why can't it hold a link reference?  Can this clarify?  Or point to another section if clarified elsewhere?

Reply:
It depends how it is tracked. My mental model for unlink was that it was based on explicit dependencies; i.e. those that are expressed in the form of a dependency manifest. There are many issues with version-range dependencies, not the least of which is determining whether removing one component will break any of your dependencies, particularly if it is permitted to have multiple versions of a single component installed (see component slots).

Having worked on a similar problem in the past, I believe that this actually becomes a rather difficult graph theory problem.

Reply:
I guess it depends on the semantics.  If the command sequence is executed when the reference count hits 0, then it's an Uninstall command sequence.  If the command sequence drops the refcount by one and then proceeds with the rest only if the refcount hits 0, then it's an Unlink command sequence.

Reply:
From what I understood of the requirements from TEEP, the intent was that this is the command sequence that would be executed **when** the refcount hits zero, not before. If that's the case, then it is certainly "uninstall." Embedding refcount semantics in the manifest processor breaks the abstraction that no state is held between invocations of the manifest processor.

My mental model of reference counting--up to this point--has been that it is done exclusively outside the manifest processor.

Reply:
Does it make more sense for unlink to apply to manifests, since we have a single, global "uninstall" command sequence for a manifest?

Reply:
See the discussion of SUIT_Digest in https://github.com/ietf-teep/teep-protocol/pull/261

Reply:
LGTM